velero-csi-plugin/volumesnapshot_restorer.go

```diff
@@ -31,29 +32,50 @@ type VSRestorer struct {
 	log logrus.FieldLogger
 }
 
+// AppliesTo returns information about which resources the VSRestorer action should be invoked for.
+// VSRestorer RestoreItemAction plugin's Execute function will only be invoked on items that match the returned
+// selector. A zero-valued ResourceSelector matches all resources.
 func (p *VSRestorer) AppliesTo() (velero.ResourceSelector, error) {
 	return velero.ResourceSelector{
 		IncludedResources: []string{"volumesnapshots.snapshot.storage.k8s.io"},
 	}, nil
 }
 
+// Execute allows the RestorePlugin to perform arbitrary logic with the item being restored,
```

Comment:
I would drop "allows the RestorePlugin to perform arbitrary logic" bit and just document *this* plugin's behavior. In this case, I think it would be helpful to be more specific about exactly what this plugin is doing -- something along the lines of "Execute modifies the VolumeSnapshot's spec to set its data source to be the relevant VolumeSnapshotContent object, rather than the original PVC the snapshot was created from, so it can be statically re-bound to the VSC on restore" (please wordsmith, just giving an example). Saying "correctly restore a CSI volumesnapshot" doesn't really give me much information.

Reply:
I am stealing your suggestion!